bokehjs/src/lib/models/widgets/tables/cell_formatters.ts

```diff
@@ -207,6 +207,35 @@ export class BooleanFormatter extends CellFormatter {
   }
 }
 
+export namespace CompositeFormatter {
+  export type Attrs = p.AttrsOf<Props>
+
+  export type Props = CellFormatter.Props & {
+    icon: p.Property<string> // XXX: enum
+  }
+}
+
+export interface CompositeFormatter extends CompositeFormatter.Attrs {}
+
+export class CompositeFormatter extends CellFormatter {
+  properties: CompositeFormatter.Props
+
+  constructor(attrs?: Partial<CompositeFormatter.Attrs>) {
+    super(attrs)
+  }
+
+  static init_CompositeFormatter(): void {
+
+    this.define<CompositeFormatter.Props>({
+      icon: [ p.String, 'check' ],
+    })
+  }
+
+  doFormat(_row: any, _cell: any, value: any, _columnDef: any, _dataContext: any): string {
+    return !!value ? i({class: this.icon}).outerHTML : ""
```

Comment:
This will be something like (untested):
```
let result = value
for (formatter of this.formatters) {
    result = formatter(_row, _cell, result, _columnDef, _dataContext
}
return result
```

Reply:
If I write this as below, I get no, but if I remove the `doFormat` line, I am getting scary red lines all over the place in this. Does it seem correct that it should be written as listed below? 

      doFormat(_row: any, _cell: any, value: any, _columnDef: any, _dataContext: any): string {
        let result = value
        for(formatter of this.formatters) {
            result = formatter(_row, _cell, result, _columnDef, _dataContext)
         }
        return result
       }